diff --git a/browser/components/preferences/preferences.xhtml b/browser/components/preferences/preferences.xhtml
index a1aec823d6266d69ff02aa8523d2da999a0871b9..559f2ea12e53dd19f24e7953db4388456b97f02c 100644
--- a/browser/components/preferences/preferences.xhtml
+++ b/browser/components/preferences/preferences.xhtml
@@ -46,6 +46,8 @@
   <link rel="stylesheet" href="chrome://browser/content/preferences/widgets/setting-pane.css" />
   <link rel="stylesheet" href="chrome://browser/content/preferences/widgets/setting-control.css" />
 
+#include enso-preferences-links.xhtml
+
   <link rel="localization" href="branding/brand.ftl"/>
   <link rel="localization" href="browser/browser.ftl"/>
     <!-- Used by fontbuilder.js -->
@@ -115,6 +117,11 @@
   <hbox flex="1">
 
     <vbox class="navigation">
+      <html:moz-input-search
+        id="searchInput"
+        data-l10n-id="search-input-box2"
+        data-l10n-attrs="placeholder, style"
+        />
       <!-- category list -->
       <richlistbox id="categories" data-l10n-id="category-list" data-l10n-attrs="aria-label">
         <richlistitem id="category-general"
@@ -127,6 +134,50 @@
           <label class="category-name" flex="1" data-l10n-id="pane-general-title"></label>
         </richlistitem>
 
+        <richlistitem id="category-enso-looks"
+                      class="category"
+                      value="paneEnsoLooks"
+                      helpTopic="prefs-main"
+                      data-l10n-id="category-enso-looks"
+                      data-l10n-attrs="tooltiptext"
+                      align="center">
+          <image class="category-icon"/>
+          <label class="category-name" flex="1" data-l10n-id="pane-enso-looks-title"></label>
+        </richlistitem>
+
+        <richlistitem id="category-enso-tabs-management"
+                      class="category"
+                      value="paneZenTabManagement"
+                      helpTopic="prefs-main"
+                      data-l10n-id="category-enso-workspaces"
+                      data-l10n-attrs="tooltiptext"
+                      align="center">
+          <image class="category-icon"/>
+          <label class="category-name" flex="1" data-l10n-id="pane-enso-tabs-title"></label>
+        </richlistitem>
+
+        <richlistitem id="category-enso-CKS"
+                      class="category"
+                      value="paneZenCKS"
+                      helpTopic="prefs-main"
+                      data-l10n-id="category-enso-CKS"
+                      data-l10n-attrs="tooltiptext"
+                      align="center">
+          <image class="category-icon"/>
+          <label class="category-name" flex="1" data-l10n-id="pane-enso-CKS-title"></label>
+        </richlistitem>
+
+        <richlistitem id="category-enso-marketplace"
+                      class="category"
+                      value="paneZenMarketplace"
+                      helpTopic="prefs-main"
+                      data-l10n-id="category-enso-marketplace"
+                      data-l10n-attrs="tooltiptext"
+                      align="center">
+          <image class="category-icon"/>
+          <label class="category-name" flex="1" data-l10n-id="pane-enso-marketplace-title"></label>
+        </richlistitem>
+
         <richlistitem id="category-home"
                       class="category"
                       value="paneHome"
@@ -242,11 +293,6 @@
                 <html:a href="about:policies" target="_blank" data-l10n-id="managed-notice"/>
               </hbox>
             </hbox>
-            <html:moz-input-search
-              id="searchInput"
-              data-l10n-id="search-input-box2"
-              data-l10n-attrs="placeholder, style"
-              />
           </hbox>
         </hbox>
         <vbox id="mainPrefPane">
@@ -260,6 +306,10 @@
 #include sync.inc.xhtml
 #include experimental.inc.xhtml
 #include moreFromMozilla.inc.xhtml
+#include ensoLooksAndFeel.inc.xhtml
+#include ensoKeyboardShortcuts.inc.xhtml
+#include ensoTabsManagement.inc.xhtml
+#include ensoMarketplace.inc.xhtml
         </vbox>
       </vbox>
     </vbox>
