diff --git a/browser/components/sessionstore/TabState.sys.mjs b/browser/components/sessionstore/TabState.sys.mjs
index 82721356d191055bec0d4b0ca49e481221988801..68437e6f9fa54fc75ca9e24d738e8afcd0ea22f8 100644
--- a/browser/components/sessionstore/TabState.sys.mjs
+++ b/browser/components/sessionstore/TabState.sys.mjs
@@ -85,7 +85,24 @@ class _TabState {
       tabData.groupId = tab.group.id;
     }
 
+    tabData.ensoWorkspace = tab.getAttribute("enso-workspace-id");
+    tabData.ensoSyncId = tab.getAttribute("id");
+    tabData.ensoEssential = tab.getAttribute("enso-essential") === "true";
+    tabData.pinned = tabData.pinned || tabData.ensoEssential;
+    tabData.ensoDefaultUserContextId = tab.getAttribute("ensoDefaultUserContextId");
+    tabData.zenPinnedIcon = tab.getAttribute("enso-pinned-icon");
+    tabData.ensoIsEmpty = tab.hasAttribute("enso-empty-tab");
+    tabData.ensoStaticLabel = tab.ensoStaticLabel;
+    tabData.ensoHasStaticIcon = !!tab.ensoStaticIcon;
+    tabData.ensoGlanceId = tab.getAttribute("glance-id");
+    tabData.ensoIsGlance = tab.hasAttribute("enso-glance-tab");
+    tabData._ensoPinnedInitialState = tab._ensoPinnedInitialState;
+    tabData._ensoIsActiveTab = tab._ensoContentsVisible;
+
     tabData.searchMode = tab.ownerGlobal.gURLBar.getSearchMode(browser, true);
+    if (tabData.searchMode?.source === tab.ownerGlobal.UrlbarUtils.RESULT_SOURCE.ENSO_ACTIONS) {
+      delete tabData.searchMode;
+    }
 
     tabData.userContextId = tab.userContextId || 0;
 
